### ISSUE TITLE
mod_sftp SFTPLog does not reliably record the IP address.

### ISSUE THREAD

**Issue by severach** [proftpd/proftpd#1244]
mod_sftp SFTPLog does not reliably record the IP address.

The first log line is "sent server version". This line does not report the IP address. The IP address is reported on "disconnecting ". Some disconnect methods don't record the IP address leaving many sessions without any IP record.

Please add or modify a SFTPLog line that always reports the IP address, preferrably at the beginning of the connection.

````
# proftpd -V
Compile-time Settings:
  Version: 1.3.8rc1 (git)
  Platform: LINUX [Linux 5.10.39-1-lts x86_64]
  Built: Fri Jun 4 2021 19:02:20 UTC
````

**Comment by Castaglia** [comment]
Does the normal ProFTPD logging (_e.g._ via `SystemLog`) have the information you need?  That is, I am wondering why this information would need to be specific to the `SFTPLog`...

**Comment by severach** [comment]
For the firewall I need a list of usernames and IP connected from. For contacting users to get them to upgrade their software to get rid of insecure encryptions I need a list of usernames, IP connected from, and encryption attempted or used.

Matching between two log files complicates log parsing far beyond the capabilities of a small shell script.

SystemLog only records the IP on an error or warning. The only two I got were LDAP and an invalid identity key. I [disabled identities](https://serverfault.com/questions/139870/stop-ssh-client-from-offering-all-the-public-keys-it-can-find) and compiled without LDAP and now SystemLog shows no log entries on a successful sftp login.

**Comment by Castaglia** [comment]
To see the "connect" messages in the `SystemLog`, you may need to increase the verbosity, using _e.g._:
```
SyslogLevel info
```

**Comment by severach** [comment]
There is no data recorded in all logs that would allow matching across them. Guessing won't work on high volume server.

This issue seems fixed in 6c5bd3c0462f701e80a3247eb7c4c3b60321d4bb
